test/e2e/snaps/ppom-blockaid-alert.spec.js

```diff
@@ -0,0 +1,192 @@
+            ],
+          },
+        ];
+
+        for (const config of testBenignConfigs) {
+          const { logExpectedDetail, method, params } = config;
+
+          // Send JSON-RPC request
+          const request = JSON.stringify({
+            jsonrpc: '2.0',
+            method,
+            params,
+          });
+          await driver.executeScript(
+            `window.transactionHash = window.ethereum.request(${request})`,
+          );
+
+          // Wait for confirmation pop-up
+          const windowHandles = await getWindowHandles(driver, 3);
+          await driver.switchToWindow(windowHandles.popup);
```

Comment:
it seems that there is an issue in the way windows are handled (some race condition that does not allow to switch to the correct window title).
I've seen that changing this code to this one, the issue is fixed for Chrome, so test is always passing now:

```
await driver.waitUntilXWindowHandles(3);
const windowHandles = await getWindowHandles(driver, 3);
 await driver.switchToWindowWithTitle('MetaMask Notification');
```

Reply:
I haven't gone deeper on how this differs from the other approach but can be investigated separately

Reply:
hm.. thanks for this! updated https://github.com/MetaMask/metamask-extension/pull/21029/commits/221bea7fe80da60ddda3007c763db026fced61f7